src/Symfony/Component/Form/Tests/Extension/Metadata/Type/MetadataTypeTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Extension\Metadata\Type;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Form\Extension\Metadata\Type\MetadataType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\Metadata\FieldMetadataInterface;
+use Symfony\Component\Form\Metadata\FormMetadataInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class MetadataTypeTest extends TestCase
+{
+    public function testGetParent()
+    {
+        ($metadata = $this->createMock(FormMetadataInterface::class))
+            ->expects($this->once())
+            ->method('getParent')
+            ->willReturn('Foo');
+
+        $this->assertEquals('Foo', (new MetadataType($metadata))->getParent());
+    }
+
+    public function testConfigureOptions()
+    {
+        ($metadata = $this->createMock(FormMetadataInterface::class))
+            ->expects($this->once())
+            ->method('getOptions')
+            ->willReturn([
+                'label' => 'Foo',
+            ]);
+
+        ($resolver = $this->createMock(OptionsResolver::class))
+            ->expects($this->once())
+            ->method('setDefaults')
+            ->with([
+                'label' => 'Foo',
+            ]);
```

Comment:
We could create an instance of `OptionResolver` and make assertions later instead. It will ensure that only `setDefaults` is called.

```php
// Arrange
$resolver = new OptionResolver();

// Act
(new MetadataType($metadata))->configureOptions($resolver);

// Assert
$this->assertSame(['label' => 'Foo'], $resolver->resolve());
```

Reply:
Done